src/test/peerman_tests.cpp

```diff
@@ -0,0 +1,188 @@
+    using NodeFn = std::function<void(CNode*)>;
+    void ForEachNode(const NodeFn& func) override {};
+    void ForEachNode(const NodeFn& func) const override {};
+    void PushMessage(CNode* pnode, CSerializedNetMsg&& msg) override;
+    CSipHasher GetDeterministicRandomizer(uint64_t id) const override { return {0, 0}; };
+    void WakeMessageHandler() override {};
+    bool OutboundTargetReached(bool historicalBlockServingLimit) const override { return true; };
+    std::vector<CAddress> GetAddresses(size_t max_addresses,
+                                       size_t max_pct,
+                                       std::optional<Network> network) const override { return {}; };
+    std::vector<CAddress> GetAddresses(CNode& requestor, size_t max_addresses, size_t max_pct) override { return {}; };
+    bool DisconnectNode(const CNetAddr& addr) override { return true; };
+    unsigned int GetReceiveFloodSize() const override { return 0; };
+    int GetExtraFullOutboundCount() const override { return 0; };
+    int GetExtraBlockRelayCount() const override { return 0; };
+    void SetTryNewOutboundPeer(bool flag) override {};
+    bool GetTryNewOutboundPeer() const override { return true; };
+    bool GetNetworkActive() const override { return true; };
+    bool GetUseAddrmanOutgoing() const override { return true; };
+    void StartExtraBlockRelayPeers() override {};
+    bool ShouldRunInactivityChecks(const CNode& node, std::chrono::seconds now) const override { return true; };
+    bool CheckIncomingNonce(uint64_t nonce) override { return true; };
+
+    void NodeReceiveMsgBytes(CNode& node, Span<const uint8_t> msg_bytes, bool& complete) const;
+    bool ReceiveMsgFrom(CNode& node, CSerializedNetMsg& ser_msg) const;
+
+    virtual ~ConnectionsInterfaceMock() {}
+
+    /** Count of number of each message type sent */
+    std::map<std::string, uint64_t> m_message_types_sent;
+};
+
+void ConnectionsInterfaceMock::PushMessage(CNode* pnode, CSerializedNetMsg&& msg)
+{
+    BOOST_TEST_MESSAGE(strprintf("sending message %s to peer %d", msg.m_type, pnode->GetId()));
+    m_message_types_sent[msg.m_type]++;
+}
+
+void ConnectionsInterfaceMock::NodeReceiveMsgBytes(CNode& node, Span<const uint8_t> msg_bytes, bool& complete) const
+{
+    assert(node.ReceiveMsgBytes(msg_bytes, complete));
+    if (complete) {
+        size_t nSizeAdded = 0;
+        auto it(node.vRecvMsg.begin());
+        for (; it != node.vRecvMsg.end(); ++it) {
+            // vRecvMsg contains only completed CNetMessage
+            // the single possible partially deserialized message are held by TransportDeserializer
+            nSizeAdded += it->m_raw_message_size;
+        }
+        {
+            LOCK(node.cs_vProcessMsg);
+            node.vProcessMsg.splice(node.vProcessMsg.end(), node.vRecvMsg, node.vRecvMsg.begin(), it);
+            node.nProcessQueueSize += nSizeAdded;
+        }
+    }
```

Comment:
#27257 dedupes this code across the entire code base, reviewing that would help move this forward.